spacemesh/v2alpha1/tx.proto

```diff
@@ -59,4 +59,106 @@ message TransactionV1 {
 
 message Nonce {
   uint64 counter = 1;
+}
+
+// TransactionState is the "journey" of a tx from mempool to block inclusion to
+// mesh to STF processing. To know whether or not the tx actually succeeded,
+// and its side effects, check the Receipt in the GlobalStateService.
+message TransactionState {
+  bytes id = 1;
+  enum TransactionState {
+    TRANSACTION_STATE_UNSPECIFIED = 0; // default state
+    TRANSACTION_STATE_REJECTED = 1; // rejected from mempool due to, e.g., invalid syntax
+    TRANSACTION_STATE_INSUFFICIENT_FUNDS = 2; // rejected from mempool by funds check
+    TRANSACTION_STATE_CONFLICTING = 3; // rejected from mempool due to conflicting counter
+    TRANSACTION_STATE_MEMPOOL = 4; // in mempool but not on the mesh yet
+    TRANSACTION_STATE_MESH = 5; // submitted to the mesh
+    TRANSACTION_STATE_PROCESSED = 6; // processed by STF; check Receipt for success or failure
+  }
+  TransactionState state = 2;
+}
+
+message TransactionResult {
+  enum Status {
+    SUCCESS = 0;
```

Comment:
please add `UNSPECIFIED` as zero, and let's use the ordinary enum naming syntax, e.g., `TRANSACTION_STATUS_UNSPECIFIED`

Reply:
Fixed

Reply:
@lrettig 
>  and let's use the ordinary enum naming syntax, e.g., TRANSACTION_STATUS_UNSPECIFIED

Why use this prefix? It leads to auto-generated constants with redundant info in their name like: `TransactionState_TRANSACTION_STATE_UNSPECIFIED` instead of `TransactionState_UNSPECIFIED`.

Reply:
Regarding `UNSPECIFIED` its part of the best practices for protobuf: https://protobuf.dev/programming-guides/dos-donts/#unspecified-enum

It is also recommended to prefix enums that belong together with the string: https://protobuf.dev/programming-guides/style/#enums I haven't found an explanation but I assume it is to avoid name collisions in some protobuf generators or assigning one enum value to a different enum (in languages that have no strict typing like Python or JS).